contrib/cirrus/cache_release.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+source $(dirname $0)/lib.sh
+
+req_env_var GOSRC CIRRUS_BUILD_ID
+
+cd $GOSRC
+
+make release_info.txt
+FIELDS="$RELEASE_BASENAME $RELEASE_VERSION $RELEASE_DIST $RELEASE_DIST_VER $RELEASE_ARCH $RELEASE_FILE"
+cat release_info.txt | read "$FIELDS"
+req_env_var $FIELDS
+
+if [[ "$CI" == "true" ]]
+then
+    req_env_var CIRRUS_BRANCH CIRRUS_TASK_NAME CIRRUS_PR
+
+    # Clearly identify source PR or branch
+    if [[ "$CIRRUS_BRANCH" != "master" ]]
```

Comment:
Readability: it's more maintainable if like things go together:
```bash
    if [[ "$CIRRUS_BRANCH" == "master" ]]; then
        NAME_SUFFIX=$CIRRUS_BRANCH
```

Reply:
Actually, that was intended to operate in an entirely different and non-obvious way...okay, I fixed it to be both more clear and maintainable...